lib/spack/spack/__init__.py

```diff
@@ -188,7 +188,7 @@
 __all__ += spack.util.executable.__all__
 
 # User's editor from the environment
-editor = Executable(os.environ.get("EDITOR", "vi"))
+editor = Executable(os.environ.get("EDITOR", "vim"))
```

Comment:
`vi` is not present on newer distributions like Fedora, and when both are present, `vi` generally enables legacy mode. How common are distributions that only come with `vi`?